Receptai.client/pages/user/dashboard/index.vue

```diff
@@ -1,13 +1,133 @@
 <script setup lang="ts">
-
+import { ref, onMounted } from 'vue';
+import axios from 'axios';
+import Widget from "./Widgets/WidgetComponent.vue";
+import RecipeBannerComponent from '~/components/Banner/RecipeBannerComponent.vue';
 definePageMeta({
   layout: "admin",
   middleware: "auth",
 });
+
+interface Recipe {
+  id: number;
+  name: string;
+  shortDescription: string;
+  author: Author;
+  dateCreated: string;
+  dateModified: string | null;
+  previewImage: string;
+  tutorialVideo?: string;
+  ingredients: Ingredients[];
+  instructions: string[];
+  tags: string[];
+  categories: Category[];
+  minutesToPrepare: number;
+  portions: number;
+}
+
+interface Author {
+  id: number;
+  name: string;
+}
+
+interface Ingredients {
+  purpose: string;
+  ingredients: Ingredient[];
+}
+
+interface Ingredient {
+  name: string;
+  quantity: number;
+  unit: string;
+}
+
+interface Category {
+  id: number;
+  name: string;
+  description: string | null;
+  previewImageUrl: string | null;
+}
+
+interface UserCookie {
+  token: string;
+  expiresIn: number;
+  user: User;
+}
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+}
+
+const currentTime = ref(new Date().toLocaleTimeString());
+
+const config = useRuntimeConfig();
+const TastyBytes_user = useCookie<UserCookie>('TastyBytes_user');
+const recipeList = ref<Recipe[]>([]);
+
+async function fetchRecipes() {
+  if (!TastyBytes_user.value?.token) {
+    console.error('User is not logged in or no token available');
+    return;
+  }
+
+  try {
+    const response = await axios.get(`${config.public.baseURL}/api/v1/recipe/list?page=0&sortBy=dateCreated&sortAsc=true`, {
+      headers: {
+        Authorization: `Bearer ${TastyBytes_user.value.token}`
+      }
+    });
+    // Filter the recipes client-side based on the logged-in user ID
+    recipeList.value = response.data.elements.filter((recipe: Recipe) => recipe.author.id === TastyBytes_user.value.user.id);
+  } catch (error) {
+    console.error("Error fetching recipes", error);
+  }
+}
+
+onMounted(fetchRecipes);
```

Comment:
I believe, that you need to use a callback for this to work correctly: https://vuejs.org/api/composition-api-lifecycle